### PR TITLE
Backfilling jinja filter tests

### PR DESCRIPTION
**Pull request by benjefferies**
### What is the context of this PR?
[trello](https://trello.com/c/csuYulm1/99-tests-for-jinja-filters)
Adding missing tests for jinja filters

### How to review 
All tests pass

### CODE REVIEW

File: tests/app/test_jinja_filters.py [hunk 1]
```diff
@@ -10,6 +12,37 @@
...
+    def test_new_line_to_break(self):
+        # Given
+        new_line = 'this is on a\nnew line'
+        context = Mock()
+        context.autoescape = False
+
+        # When
+        format_value = nl2br(context, new_line)
+
+        self.assertEquals(format_value, '<p>this is on a<br>\nnew line</p>')
```
Comment (jonnyshaw89) [inline tests/app/test_jinja_filters.py]: The name of this filter implies that is replaces the \n with a <br>?
But this test checks that there is a <br> before the \n

File: tests/app/test_jinja_filters.py [hunk 2]
```diff
@@ -10,6 +12,37 @@
...
+    def test_new_line_to_break_auto_escape(self):
+        # Given
+        new_line = '<'
+        context = Mock()
+        context.autoescape = True
+
+        # When
+        format_value = nl2br(context, new_line)
+
+        self.assertEquals(str(format_value), '<p>&lt;</p>')
```
Comment (jonnyshaw89) [inline tests/app/test_jinja_filters.py:77]: the adding of <p> tags also seems a bit implied

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: tests/app/test_jinja_filters.py (hunk 1 above)
Comment (benjefferies) [reply tests/app/test_jinja_filters.py]: I don't think it has any impact but is confusing so will change the implementation

File: tests/app/test_jinja_filters.py (hunk 2 above)
Comment (benjefferies) [reply tests/app/test_jinja_filters.py:77]: Is it clearer with the new method name?

Comment (jonnyshaw89) [reply tests/app/test_jinja_filters.py:77]: Much

**Review by jonnyshaw89 — APPROVED** [review]

File: app/jinja_filters.py [hunk 3]
```diff
@@ -21,9 +21,9 @@ def format_percentage(value):
 
 @evalcontextfilter
 @blueprint.app_template_filter()
-def nl2br(context, value):
+def format_multilined_string(context, value):
     _paragraph_re = re.compile(r'(?:\r\n|\r|\n){2,}')
-    result = u'\n\n'.join(u'<p>%s</p>' % p.replace('\n', '<br>\n') for p in _paragraph_re.split(escape(value)))
+    result = u'\n\n'.join(u'<p>%s</p>' % p.replace('\n', '<br>') for p in _paragraph_re.split(escape(value)))
```
Comment (samiwel) [inline app/jinja_filters.py]: It might be correct, but why are we joining with `u'\n\n'`? Isn't this method replacing \n with `<br>`?

Comment (jonnyshaw89) [reply app/jinja_filters.py]: i think it maintains the line breaks in the source code but also added the br so it appears correct when rendered?

Comment (jonnyshaw89) [reply app/jinja_filters.py]: So maybe my previous comment about removing the \n is incorrect?
Depends if we care how the source looks?

Comment (benjefferies) [reply app/jinja_filters.py]: As it's generated source then we probably don't care so I'll remove

File: app/jinja_filters.py [hunk 4]
```diff
@@ -21,9 +21,9 @@ def format_percentage(value):
 
 @evalcontextfilter
 @blueprint.app_template_filter()
-def nl2br(context, value):
+def format_multilined_string(context, value):
     _paragraph_re = re.compile(r'(?:\r\n|\r|\n){2,}')
-    result = u'\n\n'.join(u'<p>%s</p>' % p.replace('\n', '<br>\n') for p in _paragraph_re.split(escape(value)))
+    result = u''.join(u'<p>%s</p>' % p.replace('\n', '<br>') for p in _paragraph_re.split(escape(value)))
```
Comment (jonnyshaw89) [inline app/jinja_filters.py]: can probably remove the join completely?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/jinja_filters.py (hunk 4 above)
Comment (benjefferies) [reply app/jinja_filters.py]: I have changed this a lot now as I think there was a bug in the code

**Review by jonnyshaw89 — APPROVED** [review]

**Review by samiwel — APPROVED** [review]
LGTM

**Pull request merged**